From 5dc52db6771ac0c9bbb583d903fe2a9b64af0b96 Mon Sep 17 00:00:00 2001
From: Jacky Chou <jacky_chou@aspeedtech.com>
Date: Tue, 5 May 2026 13:30:17 +0800
Subject: [PATCH] board: aspeed: add evb-ast2705 board support

Add the EVB board files and config header for the AST2705 evaluation
board.

Upstream-Status: Backport [SDK v11.03 migration]
Signed-off-by: Jacky Chou <jacky_chou@aspeedtech.com>
Change-Id: I928dc16e8c7fc3f671436b2194447ae04cf5fe30
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 board/aspeed/evb_ast2705/Kconfig       | 13 ++++++++++
 board/aspeed/evb_ast2705/Makefile      |  1 +
 board/aspeed/evb_ast2705/evb_ast2705.c |  5 ++++
 include/configs/evb_ast2705.h          | 34 ++++++++++++++++++++++++++
 4 files changed, 53 insertions(+)
 create mode 100644 board/aspeed/evb_ast2705/Kconfig
 create mode 100644 board/aspeed/evb_ast2705/Makefile
 create mode 100644 board/aspeed/evb_ast2705/evb_ast2705.c
 create mode 100644 include/configs/evb_ast2705.h

diff --git a/board/aspeed/evb_ast2705/Kconfig b/board/aspeed/evb_ast2705/Kconfig
new file mode 100644
index 0000000000..654518d1d1
--- /dev/null
+++ b/board/aspeed/evb_ast2705/Kconfig
@@ -0,0 +1,13 @@
+if TARGET_EVB_AST2705
+
+config SYS_BOARD
+	default "evb_ast2705"
+
+config SYS_VENDOR
+	default "aspeed"
+
+config SYS_CONFIG_NAME
+	string "board configuration name"
+	default "evb_ast2705"
+
+endif
diff --git a/board/aspeed/evb_ast2705/Makefile b/board/aspeed/evb_ast2705/Makefile
new file mode 100644
index 0000000000..87284c70ac
--- /dev/null
+++ b/board/aspeed/evb_ast2705/Makefile
@@ -0,0 +1 @@
+obj-y += evb_ast2705.o
diff --git a/board/aspeed/evb_ast2705/evb_ast2705.c b/board/aspeed/evb_ast2705/evb_ast2705.c
new file mode 100644
index 0000000000..b34aa6e168
--- /dev/null
+++ b/board/aspeed/evb_ast2705/evb_ast2705.c
@@ -0,0 +1,5 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) ASPEED Technology Inc.
+ */
+
diff --git a/include/configs/evb_ast2705.h b/include/configs/evb_ast2705.h
new file mode 100644
index 0000000000..764daf4544
--- /dev/null
+++ b/include/configs/evb_ast2705.h
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
+/*
+ * Copyright (C) ASPEED Technology Inc.
+ */
+
+#ifndef __CONFIG_H
+#define __CONFIG_H
+
+#include <configs/aspeed-common.h>
+
+/* Extra ENV for Boot Command */
+#define STR_HELPER(n)	#n
+#define STR(n)		STR_HELPER(n)
+
+#define CFG_SYS_UBOOT_BASE		CONFIG_TEXT_BASE
+
+#define CFG_EXTRA_ENV_SETTINGS \
+	"bootspi=fdt addr ${fdtspiaddr} && fdt header get fitsize totalsize && " \
+	"cp.b ${fdtspiaddr} ${loadaddr} ${fitsize} && bootm ${loadaddr}; " \
+	"echo Error loading kernel FIT image\0" \
+	"loadaddr=" STR(CONFIG_SYS_LOAD_ADDR) "\0"	\
+	"bootside=a\0"	\
+	"rootfs=rofs-a\0"	\
+	"setmmcargs=setenv bootargs ${bootargs} rootwait root=PARTLABEL=${rootfs}\0"	\
+	"boota=setenv bootpart 2; setenv rootfs rofs-a; run setmmcargs; ext4load mmc 0:${bootpart} ${loadaddr} fitImage && bootm ${loadaddr}; echo Error loading kernel FIT image\0"	\
+	"bootb=setenv bootpart 3; setenv rootfs rofs-b; run setmmcargs; ext4load mmc 0:${bootpart} ${loadaddr} fitImage && bootm ${loadaddr}; echo Error loading kernel FIT image\0"	\
+	"bootmmc=if test \"${bootside}\" = \"b\"; then run bootb; run boota; else run boota; run bootb; fi\0"	\
+	"setufsargs=setenv bootargs ${bootargs} rootwait root=PARTLABEL=${rootfs}\0"	\
+	"ufsboota=setenv bootpart 2; setenv rootfs rofs-a; run setufsargs; ext4load scsi 0:${bootpart} ${loadaddr} fitImage && bootm ${loadaddr}; echo Error loading kernel FIT image\0"	\
+	"ufsbootb=setenv bootpart 3; setenv rootfs rofs-b; run setufsargs; ext4load scsi 0:${bootpart} ${loadaddr} fitImage && bootm ${loadaddr}; echo Error loading kernel FIT image\0"	\
+	"bootufs=if test \"${bootside}\" = \"b\"; then run ufsbootb; run ufsboota; else run ufsboota; run ufsbootb; fi\0"	\
+	"verify=no\0"	\
+	""
+#endif	/* __CONFIG_H */
-- 
2.43.0

